runtime/caml/memory.h

```diff
@@ -185,11 +185,12 @@ CAMLextern wchar_t* caml_stat_wcsconcat(int n, ...);
 #endif
 
 #ifdef DEBUG
-#define DEBUG_clear(result, wosize) do{ \
-  uintnat caml__DEBUG_i; \
-  for (caml__DEBUG_i = 0; caml__DEBUG_i < (wosize); ++ caml__DEBUG_i){ \
-    Field ((result), caml__DEBUG_i) = Debug_uninit_minor; \
-  } \
+#define DEBUG_clear(result, wosize) do{                                     \
+  uintnat caml__DEBUG_i;                                                    \
+  for (caml__DEBUG_i = 0; caml__DEBUG_i < (wosize); ++ caml__DEBUG_i) {     \
+    CAMLassert(Field((result), caml__DEBUG_i) == Debug_free_minor);         \
+    Field ((result), caml__DEBUG_i) = Debug_uninit_minor;                   \
+  }                                                                         \
```

Comment:
Our of curiosity: why is this a macro? It looks like a `Caml_inline` (or just `static`) function would do just fine, and avoid the weird `caml__DEBUG_i` naming convolutions or the exciting question of duplicate evaluations when `wosize` is an expression.

Reply:
I agree that `Caml_inline` would be better than a macro. I will make that change.

Reply:
Fixed in 79cfe30